diag_manager2/fms_diag_axis.F90

```diff
@@ -284,4 +291,70 @@ integer function get_axis_set_num(set_name)
     end do
 end function get_axis_set_num
 
+
+
+
+!! Register and write the information of the axis.
+!! Assumes: axes object is initalized by this point
+!! TODO : It would be convenient to store axis info inside the axis object
+!!      so that conversions (e.g. character case) are not needed for writing or testing.
+!! TODO : Other file types besides FmsNetcdfFile_t
+!! REFERENCE: See functions diag_util.F90::opening_file and diag_output.F90::write_axis_meta_data
+!!   and write_field_meta_data
+subroutine send_data_generic(this, fobj)
+    class(diag_axis_type),   intent(inout)   :: this
+    type(FmsNetcdfFile_t),  intent(inout)   :: fobj
+    class(FmsNetcdfFile_t), pointer    :: fptr
+    !!
+    select type (fptr) !! Also check X or Y axis
```

Comment:
You never point at anything.  If you want to point at `fobj`, then you need to declare it with the `target` keyword
```fortran
type(FmsNetcdfFile_t),  intent(inout), targer   :: fobj
```